src/orquestra/sdk/mlflow/_connection_utils.py

```diff
@@ -6,15 +6,111 @@
 
 import os
 import typing as t
+import warnings
 from pathlib import Path
+from typing import Optional, Tuple
 
+from requests import Response, Session
+
+from orquestra import sdk
+from orquestra.sdk._base import _env
 from orquestra.sdk._base._config import read_config
 from orquestra.sdk._base._env import CURRENT_USER_ENV
 from orquestra.sdk._base._jwt import get_email_from_jwt_token
 from orquestra.sdk._base._services import ORQUESTRA_BASE_PATH
+from orquestra.sdk._base._spaces._api import make_workspace_url, make_workspace_zri
 from orquestra.sdk.exceptions import ConfigNameNotFoundError, RuntimeConfigError
+from orquestra.sdk.schema.configs import ConfigName
 
 DEFAULT_TEMP_ARTIFACTS_DIR: Path = ORQUESTRA_BASE_PATH / "mlflow" / "artifacts"
+RESOURCE_CATALOG_URI: str = "http://orquestra-resource-catalog.resource-catalog"
+
+
+# region: private
+def _is_executing_remoteley() -> bool:
+    """
+    Determine whether the code is being executed locally, or on a cluster/studio.
+    """
+    if os.getenv(_env.CURRENT_CLUSTER_ENV):
+        return True
+    else:
+        # TODO: https://zapatacomputing.atlassian.net/browse/ORQSDK-914 - this
+        # workaround exists to maintian compatibility with previous WDR versions.
+        # Future releases should remove this else block.
+        envvars = [
+            _env.PASSPORT_FILE_ENV,
+            _env.MLFLOW_CR_NAME,
+            _env.MLFLOW_PORT,
+            _env.MLFLOW_ARTIFACTS_DIR,
+        ]
+        if None not in [os.getenv(envvar) for envvar in envvars]:
```

Comment:
**nitpick**: when reading this line I find it difficult to figure out the author's intention, there's just too many negations! WDYT about the following, does it capture the same thing?
```suggestion
        if all(os.getenv(envvar) for envvar in envvars):
```